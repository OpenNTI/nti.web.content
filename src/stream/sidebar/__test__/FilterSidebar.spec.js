import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import FilterSidebar from '../FilterSidebar';

class Sidebar extends React.Component {
	state = {
		params: {
			types: {
				'NOTES': true,
				'BOOKMARKS': true,
				'HIGHLIGHTS': true,
				'THOUGHTS': true,
				'LIKES': true
			},
			batchAfter: null,
			sortOn: 'CreatedTime',
			sortOrder: 'descending'
		}
	}

	onDateChange = option => {
		this.setState({ ...this.state.params, batchAfter: option.value });
	};

	onTypeChange = (option, selected) => {
		this.setState({
			...this.state.params,
			types: {
				...this.state.params.types,
				[option.value]: selected
			}
		});
	};

	onSortByChange = ({ target: { value } }) => {
		this.setState({ ...this.state.params, sortOn: value });
	};

	render () {
		return (
			<FilterSidebar
				params={this.state.params}
				onDateChange={this.onDateChange}
				onTypeChange={this.onTypeChange}
				onSortByChange={this.onSortByChange}
			/>
		);
	}
}

/* eslint-env jest */
describe('FilterSidebar', () => {
	test('Simple case', () => {
		const filterBar = renderer.create(
			<FilterSidebar
				onSortByChange = {() => {}}
				onDateChange = {() => {}}
				onTypeChange = {() => {}}
				params={{
					'types': {
						'NOTES': true,
						'BOOKMARKS': true,
						'HIGHLIGHTS': true,
						'THOUGHTS': true,
						'LIKES': true
					},
					'batchAfter': null,
					'sortOn': 'CreatedTime',
					'sortOrder': 'descending'
				}}
			/>
		);
		const tree = filterBar.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('Selecting New Options', () => {
		const cmp = mount(<Sidebar />);

		const threeMonths = cmp.find('.option.date-filter').at(3);
		threeMonths.simulate('click');
		cmp.update();
		expect(cmp.state('batchAfter')).toBe('pastthreemonths');

		const highlights = cmp.find('.option.type-filter.selected').at(2);
		highlights.find('input').simulate('change', { target: { checked: false } });
		cmp.update();
		expect(cmp.state('types').HIGHLIGHTS).toBe(false);

		const sortby = cmp.find('.nti-select-native.stream-sidebar-sort');
		sortby.simulate('change', { target: { value: 'LikeCount' } });
		cmp.update();
		expect(cmp.state('sortOn')).toBe('LikeCount');
	});

	test('Override Activity Type', () => {
		const typeOptions = [
			{ label: 'Thoughts', value: 'THOUGHTS' },
			{ label: 'Discussions', value: 'DISCUSSIONS' },
			{ label: 'Notes', value: 'NOTES' },
			{ label: 'Chat', value: 'CHAT' }
		];
		const cmp = renderer.create(
			<FilterSidebar
				typeOptions={typeOptions}
				onSortByChange={() => { }}
				onDateChange={() => { }}
				onTypeChange={() => { }}
				params={{
					'types': {
						'NOTES': true,
						'DISCUSSIONS': true,
						'THOUGHTS': true,
						'CHAT': true,
					},
					'batchAfter': null,
					'sortOn': 'CreatedTime',
					'sortOrder': 'descending'
				}}
			/>
		);
		const tree = cmp.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('Override Sort By Options', () => {
		const sortByOptions = [
			{ label: 'Date Created', value: 'CreatedTime' },
			{ value: 'FirstModified', label: 'Last Activity' },
			{ value: 'ReferencedByLeast', label: 'Least Commented' },
			{ value: 'CountLike', label: 'Liked Most' }
		];
		const cmp = renderer.create(
			<FilterSidebar
				sortByOptions={sortByOptions}
				onSortByChange={() => { }}
				onDateChange={() => { }}
				onTypeChange={() => { }}
				params={{
					'types': {
						'NOTES': true,
						'DISCUSSIONS': true,
						'THOUGHTS': true,
						'CHAT': true,
					},
					'batchAfter': null,
					'sortOn': 'CreatedTime',
					'sortOrder': 'descending'
				}}
			/>
		);
		const tree = cmp.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
