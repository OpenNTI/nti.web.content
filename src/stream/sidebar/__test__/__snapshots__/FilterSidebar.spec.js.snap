// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSidebar Override Activity Type 1`] = `
<div
  className="filter-sidebar"
>
  <div
    className="select-title"
  >
    SORT BY
  </div>
  <div
    className="nti-select-native-wrapper"
  >
    <span
      className="icon-chevron-down small"
      role="presentation"
    />
    <span
      className="nti-select-value"
      role="presentation"
    >
      Date Created
    </span>
    <select
      className="nti-select-native stream-sidebar-sort"
      onChange={[Function]}
      value="CreatedTime"
    >
      <option
        value="CreatedTime"
      >
        Date Created
      </option>
      <option
        value="LastModified"
      >
        Recent Activity
      </option>
      <option
        value="ReferencedByCount"
      >
        Most Commented
      </option>
      <option
        value="LikeCount"
      >
        Most Liked
      </option>
    </select>
  </div>
  <div
    className="stream-date-range"
  >
    <div
      className="option-title"
    >
      Date Range
    </div>
    <div
      className="option date-filter selected"
      onClick={[Function]}
    >
      Anytime
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past Week
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past Month
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past 3 Months
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past Year
    </div>
  </div>
  <div
    className="stream-type-filter"
  >
    <div
      className="option-title"
    >
      ACTIVITY TYPE
    </div>
    <div
      className="option type-filter selected"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Thoughts
      </span>
    </div>
    <div
      className="option type-filter selected"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Discussions
      </span>
    </div>
    <div
      className="option type-filter selected"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Notes
      </span>
    </div>
    <div
      className="option type-filter selected"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Chat
      </span>
    </div>
  </div>
</div>
`;

exports[`FilterSidebar Override Sort By Options 1`] = `
<div
  className="filter-sidebar"
>
  <div
    className="select-title"
  >
    SORT BY
  </div>
  <div
    className="nti-select-native-wrapper"
  >
    <span
      className="icon-chevron-down small"
      role="presentation"
    />
    <span
      className="nti-select-value"
      role="presentation"
    >
      Date Created
    </span>
    <select
      className="nti-select-native stream-sidebar-sort"
      onChange={[Function]}
      value="CreatedTime"
    >
      <option
        value="CreatedTime"
      >
        Date Created
      </option>
      <option
        value="FirstModified"
      >
        Last Activity
      </option>
      <option
        value="ReferencedByLeast"
      >
        Least Commented
      </option>
      <option
        value="CountLike"
      >
        Liked Most
      </option>
    </select>
  </div>
  <div
    className="stream-date-range"
  >
    <div
      className="option-title"
    >
      Date Range
    </div>
    <div
      className="option date-filter selected"
      onClick={[Function]}
    >
      Anytime
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past Week
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past Month
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past 3 Months
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past Year
    </div>
  </div>
  <div
    className="stream-type-filter"
  >
    <div
      className="option-title"
    >
      ACTIVITY TYPE
    </div>
    <div
      className="option type-filter selected"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Notes
      </span>
    </div>
    <div
      className="option type-filter"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Bookmarks
      </span>
    </div>
    <div
      className="option type-filter"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Highlights
      </span>
    </div>
    <div
      className="option type-filter"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Likes
      </span>
    </div>
  </div>
</div>
`;

exports[`FilterSidebar Simple case 1`] = `
<div
  className="filter-sidebar"
>
  <div
    className="select-title"
  >
    SORT BY
  </div>
  <div
    className="nti-select-native-wrapper"
  >
    <span
      className="icon-chevron-down small"
      role="presentation"
    />
    <span
      className="nti-select-value"
      role="presentation"
    >
      Date Created
    </span>
    <select
      className="nti-select-native stream-sidebar-sort"
      onChange={[Function]}
      value="CreatedTime"
    >
      <option
        value="CreatedTime"
      >
        Date Created
      </option>
      <option
        value="LastModified"
      >
        Recent Activity
      </option>
      <option
        value="ReferencedByCount"
      >
        Most Commented
      </option>
      <option
        value="LikeCount"
      >
        Most Liked
      </option>
    </select>
  </div>
  <div
    className="stream-date-range"
  >
    <div
      className="option-title"
    >
      Date Range
    </div>
    <div
      className="option date-filter selected"
      onClick={[Function]}
    >
      Anytime
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past Week
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past Month
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past 3 Months
    </div>
    <div
      className="option date-filter"
      onClick={[Function]}
    >
      Past Year
    </div>
  </div>
  <div
    className="stream-type-filter"
  >
    <div
      className="option-title"
    >
      ACTIVITY TYPE
    </div>
    <div
      className="option type-filter selected"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Notes
      </span>
    </div>
    <div
      className="option type-filter selected"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Bookmarks
      </span>
    </div>
    <div
      className="option type-filter selected"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Highlights
      </span>
    </div>
    <div
      className="option type-filter selected"
    >
      <label
        className="checkbox-component"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="label"
        />
      </label>
      <span
        onClick={[Function]}
      >
        Likes
      </span>
    </div>
  </div>
</div>
`;
